.github/workflows/nightly-build.yml

```
@@ -0,0 +1,183 @@
+name: Nightly Upstream Snapshot Build
+description: This workflow checks for new upstream versions of OpenTelemetry dependencies, creates a PR to update them, and builds and tests the new changes.
+
+on:
+  schedule:
+    - cron: "21 3 * * *"
+  workflow_dispatch:
+  push:
+    branches:
+      - zhaez/nightly-build
+
+env:
+  AWS_DEFAULT_REGION: us-east-1
+  BRANCH_NAME: nightly-dependency-updates
+
+permissions:
+  contents: write
+  pull-requests: write
+  id-token: write
+
+jobs:
+  update-dependencies:
+    runs-on: ubuntu-latest
+    outputs:
+      has_changes: ${{ steps.check_changes.outputs.has_changes }}
+      otel_js_core_version: ${{ steps.get_versions.outputs.otel_js_core_version }}
+      otel_js_experimental_version: ${{ steps.get_versions.outputs.otel_js_experimental_version }}
+      otel_js_api_version: ${{ steps.get_versions.outputs.otel_js_api_version }}
+      otel_js_semconv_version: ${{ steps.get_versions.outputs.otel_js_semconv_version }}
+      breaking_changes_info: ${{ steps.breaking_changes.outputs.breaking_changes_info }}
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 #v5.0.0
+      with:
+        fetch-depth: 0
+        token: ${{ secrets.GITHUB_TOKEN }}
+    
+    - name: Set up Node.js
+      uses: actions/setup-node@a0853c24544627f65ddf259abe73b1d18a591444 #v5.0.0
+      with:
+        node-version: '18'
+    
+    - name: Get latest upstream versions
+      id: get_versions
+      run: node scripts/get_upstream_versions.js
+    
+    - name: Check for breaking changes
+      id: breaking_changes
+      run: node scripts/find_breaking_changes.js
+    
+    - name: Configure git and create branch
+      run: |
+        git config --local user.name "github-actions"
+        git config --local user.email "github-actions@github.com"
```

Please keep consistent across lang.
```suggestion
    - name: Setup Git
      run: |
        git config user.name "github-actions"
        git config user.email "github-actions@github.com"
```